holoviz/src/main.rs

```diff
@@ -26,21 +29,110 @@ struct Point {
     y: f32,
 }
 
+/// SVG viewBox extents
+#[derive(PartialEq, Debug, Copy, Clone)]
+struct Extents {
+    xmin: f32,
+    ymin: f32,
+    xmax: f32,
```

Comment:
I like this `Extent` abstraction! 

Functionally, I don't think this makes any difference in your code, but it did trip me up for while when I was trying to get objects with negative coordinates centered: the second two arguments in the extent are actually `width` and `height`.